pull_target.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/phrase/phraseapp-go/phraseapp"
+)
+
+type Targets []*Target
+
+func (targets Targets) ProjectIds() []string {
+	projectIds := []string{}
+	for _, target := range targets {
+		projectIds = append(projectIds, target.ProjectID)
+	}
+	return projectIds
+}
+
+type Target struct {
+	File          string
+	ProjectID     string
+	AccessToken   string
+	FileFormat    string
+	Params        *PullParams
+	RemoteLocales []*phraseapp.Locale
+}
+
+func (target *Target) CheckPreconditions() error {
+	if err := ValidPath(target.File, target.FileFormat, ""); err != nil {
+		return err
+	}
+
+	preconditions := []func(*Target) error{
+		containsStars,
+		containsDuplicatePlaceholders,
+		containsAmbiguousLocaleInformation,
+		containsInvalidTagInformation,
+	}
+
+	for _, precondition := range preconditions {
+		if err := precondition(target); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func containsStars(target *Target) error {
+	if strings.Count(target.File, "*") > 0 {
```

Comment:
there is also `strings.Contains` which is (probably) more efficient (as it only looks for the first match)